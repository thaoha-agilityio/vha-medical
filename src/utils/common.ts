import { STATUS } from '@/components/ui';
import {
  APIRelatedResponse,
  APIResponse,
  Option,
  RolePermission,
  SpecialtyResponse,
  UserLogged,
  UserModel,
} from '@/types';

export const getObjectValue = <T, Key extends keyof T>(obj: T, key: string) => {
  return obj[key as Key] as string;
};

export const getDescriptionNotification = ({
  userId,
  senderId,
  content,
  time,
}: {
  userId: string;
  senderId: APIRelatedResponse<APIResponse<UserModel>>;
  content: string;
  time: string;
}) => {
  const { id = '', attributes } = senderId.data || {};
  const { username = '' } = attributes || {};

  const name = userId == id ? 'You' : username;
  return `${name} ${content} at ${time}`;
};

export const getGreeting = (): string => {
  const now = new Date();
  const currentHour = now.getHours();

  switch (true) {
    case currentHour < 12:
      return 'Good Morning';
    case currentHour < 18:
      return 'Good Afternoon';
    default:
      return 'Good Evening';
  }
};

export const getStatusKey = (value: string) => {
  const entry = Object.entries(STATUS).find(([_, val]) => val === value);
  return entry ? parseInt(entry[0]) : undefined;
};

export const transformUsers = (users: UserLogged[]): Option[] =>
  users.map(({ id, email }) => ({
    key: id,
    label: email,
  }));

export const formatString = (input: string) =>
  input.toLowerCase().replace(/\s+/g, '_');

export const transformSpecialtiesByName = (
  specialties: SpecialtyResponse[],
): Option[] => [
  { key: 'all', label: 'All' },
  ...specialties.map((specialty) => ({
    key: formatString(specialty.attributes.name),
    label: specialty.attributes.name,
  })),
];

export const formatSpecialtyString = (input: string | undefined) => {
  if (!input) return '';
  return input.replaceAll('_', ' ');
};

export const getRoleIdByName = (roles: RolePermission[], roleName: string) =>
  roles.find((role) => role.name === roleName)?.id;

export const transformSpecialtiesById = (
  specialties: SpecialtyResponse[],
): Option[] =>
  specialties.map((specialty) => ({
    key: specialty.id,
    label: specialty.attributes.name,
  }));
