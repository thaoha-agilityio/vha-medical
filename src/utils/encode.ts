import { ENCODE_COOKIE } from '@/constants';
import crypto from 'crypto';

export const encrypt = async (value: string) => {
  const iv = crypto.randomBytes(16);

  const secretKeyString = ENCODE_COOKIE.secret || '';

  if (!secretKeyString) return;

  const secretKey = Buffer.from(secretKeyString, 'base64');

  const cipher = crypto.createCipheriv(ENCODE_COOKIE.algorithm, secretKey, iv);

  let encrypted = cipher.update(value, 'utf8', 'hex');
  encrypted += cipher.final('hex');

  return `${iv.toString('hex')}:${encrypted}`;
};

export const decrypt = async (encrypted: string) => {
  const [ivHex, encryptedData] = encrypted.split(':');

  const secretKeyString = ENCODE_COOKIE.secret || '';

  if (!secretKeyString) return;

  const secretKey = Buffer.from(secretKeyString, 'base64');

  const decipher = crypto.createDecipheriv(
    ENCODE_COOKIE.algorithm,
    secretKey,
    Buffer.from(ivHex, 'hex'),
  );

  let decrypted = decipher.update(encryptedData, 'hex', 'utf8');
  decrypted += decipher.final('utf8');

  return decrypted;
};
