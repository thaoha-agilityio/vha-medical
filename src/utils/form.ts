import {
  FieldErrors,
  FieldValues,
  Path,
  UseFormClearErrors,
} from 'react-hook-form';

/**
 * Clear error message when the users typing
 */
export const clearErrorOnChange = <T extends FieldValues>(
  fieldName: Path<T>,
  errors: FieldErrors<T>,
  clearErrorFunc: UseFormClearErrors<T>,
): void => {
  errors[fieldName]?.message && clearErrorFunc(fieldName);
};

/**
 * Enable submission when all required fields are filled and there are no errors
 */
export const isEnableSubmit = (
  requiredFields: string[],
  dirtyFields: string[],
  errors: Record<string, unknown>,
): boolean => {
  const isMatchAllRequiredFields = requiredFields.every((field) =>
    dirtyFields.includes(field),
  );
  return isMatchAllRequiredFields && errors && !Object.keys(errors).length;
};
