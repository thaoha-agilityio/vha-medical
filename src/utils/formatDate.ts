import { Option } from '@/types';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import relativeTime from 'dayjs/plugin/relativeTime';

dayjs.extend(relativeTime);

/**
 * @param value string - Date time string to format
 * @returns string - Formatted date time
 */
export const formatDateTime = (value: string) => {
  return dayjs(value).format('MMM D, YYYY h:mm a');
};

/**
 * @param value string - Date string to format
 * @returns string - Formatted date
 */
export const formatNewDate = (value: string) => {
  return dayjs(value).format('MMM D, YYYY');
};

/**
 * @param start string - Start time string to format
 * @param duration string - Duration time string to format
 * @returns string - Formatted date
 */
export const formatTimeAppointment = ({
  start,
  duration,
}: {
  start: string;
  duration: string;
}) => {
  const startTime = dayjs(start);
  const [hours, minutes, _] = duration.split(':').map(Number);
  const endTime = startTime.add(hours, 'hour').add(minutes, 'minute');

  return `${startTime.format('h:mm a')} to ${endTime.format('h:mm a')}`;
};

/**
 * Formats the input date string into an abbreviated day of the week and the day of the month.
 * @param input - The input date string in ISO 8601 format ('2024-09-11T06:30:00.000Z').
 * @returns An object containing:
 *   - dayOfWeek: Wed.
 *   - dayOfMonth: 11.
 */
export const formatDate = (input: string) => {
  const date = dayjs(input);
  dayjs.extend(advancedFormat);

  // Get the abbreviated day of the week
  const dayOfWeek = date.format('ddd');

  // Get the day of the month
  const dayOfMonth = date.format('DD');

  return { dayOfWeek, dayOfMonth };
};

/**
 * Format a date string from ISO string to Day Month Year.
 * @param {string} isoString - ISO string
 * @returns {string} Formatted date string
 */
export const formatDayMonthYear = (isoString: string) =>
  dayjs(isoString).format('D MMM YYYY');

/**
 * Converts an ISO date string to a relative time string.
 *
 * @param {string} isoString - The ISO date string.
 * @returns {string} - The relative time from the given date to now.
 *
 * @example
 * fromDateToNow('2024-09-20T14:00:00Z'); // "1 day ago"
 */
export const fromDateToNow = (isoString: string): string =>
  dayjs(isoString).fromNow();

export const isLaterThanCurrentTime = (isoString: string) => {
  const inputDate = dayjs(isoString);
  const currentDate = dayjs();

  return inputDate.isAfter(currentDate);
};

export const generateTimeOptions = (): Option[] => {
  const times: Option[] = [];
  const hour = 5;
  for (let i = 1; i <= hour * 4; i++) {
    const hour = Math.floor(i / 4);
    const minutes = (i % 4) * 15;

    const totalMinutes = hour * 60 + minutes;

    // Format the hours and minutes to be two digits
    const hourStr = hour.toString().padStart(2, '0');
    const minuteStr = minutes.toString().padStart(2, '0');

    // Push the new object into the time array
    times.push({
      key: totalMinutes.toString(),
      label: `${hourStr}:${minuteStr}`,
    });
  }

  return times;
};
