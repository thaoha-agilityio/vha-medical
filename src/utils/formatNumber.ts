/**
 * Formats a given number using compact notation (e.g., "1.2K", "1M") and appends a unit with proper pluralization.
 *
 * @param {number} number - The number to format. Will be shortened using compact notation.
 * @param {string} [unit=''] - The unit to append after the formatted number. Will be pluralized if the number is greater than 1.
 *
 * @returns {string} - A string that combines the formatted number and the unit (with correct pluralization).
 *
 * @example
 * // Returns "1.2K items"
 * formatNumberWithUnit(1200, 'item');
 *
 * @example
 * // Returns "1 watch"
 * formatNumberWithUnit(1, 'watch');
 *
 * @example
 * // Returns "1.5M users"
 * formatNumberWithUnit(1500000, 'user');
 */
export const formatNumberWithUnit = (number: number, unit: string = '') => {
  return `${new Intl.NumberFormat('en', {
    notation: 'compact',
    compactDisplay: 'short',
    maximumFractionDigits: 1,
  }).format(
    number,
  )} ${Math.abs(number) > 1 && unit ? `${unit}s` : `${unit}`}`.replace(
    '.',
    ',',
  );
};
