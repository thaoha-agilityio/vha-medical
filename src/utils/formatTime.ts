import dayjs from 'dayjs';
import { TimeInputValue } from '@nextui-org/react';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';

dayjs.extend(timezone);
dayjs.extend(utc);

export const formatStartDate = (isoString: string) => {
  return dayjs(isoString).format('YYYY-MM-DD');
};

export const convertToTimeObject = (isoString: string) => {
  if (!isoString) return undefined;

  const startTime = dayjs(isoString);

  return {
    hour: startTime.hour(),
    minute: startTime.minute(),
    second: startTime.second(),
    millisecond: startTime.millisecond(),
  };
};

/**
 * Generates an ISO date string from a TimeInputValue object.
 *
 * @param inputTime - The TimeInputValue object containing the time values.
 * @param dateTime - The date string in the format "YYYY-MM-DD".
 * */
export const generateISODate = (
  inputTime: TimeInputValue,
  dateTime: string,
) => {
  const { hour = 0, minute = 0, second = 0, millisecond = 0 } = inputTime;

  // Parse the provided date string with dayjs in UTC mode
  const date = dayjs(dateTime);

  // Set the time using the input object
  const updatedDate = date
    .hour(hour)
    .minute(minute)
    .second(second)
    .millisecond(millisecond);

  // Return the ISO string
  return updatedDate.toISOString();
};

export const convertMinutesToTime = (totalMinutes: string) => {
  // Create a dayjs object at the start of the epoch (00:00:00)
  const start = dayjs().startOf('day');

  // Add the total minutes to the start
  const time = start.add(+totalMinutes, 'minute');

  // Format the time string
  return time.format('HH:mm:ss');
};

export const convertTimeToMinutes = (time: string) => {
  // Split the time by colon to extract hours and minutes
  const [hours, minutes] = time.split(':');

  // Convert hours and minutes to integers
  const totalHours = parseInt(hours, 10);
  const totalMinutes = parseInt(minutes, 10);

  // Calculate the total minutes
  return totalHours * 60 + totalMinutes;
};

/**
 * Checks if a given date is in the future relative to the current time in GMT+7 timezone (Asia/Bangkok).
 *
 * @param {string} dateInput - The date string to be checked.
 * @returns {boolean} - Returns true if the input date is in the future, false otherwise.
 */
export const isFutureDate = (dateInput: string) => {
  // GMT+7 timezone
  const currentTime = dayjs().tz('Asia/Bangkok');

  return dayjs(dateInput).tz('Asia/Bangkok').isAfter(currentTime);
};
