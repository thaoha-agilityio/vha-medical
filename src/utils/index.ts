export * from './common';
export * from './form';
export * from './formatDate';
export * from './formatTime';
export * from './styles';
export * from './formatNumber';
