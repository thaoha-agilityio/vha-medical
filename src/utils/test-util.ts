import type { ReactElement, ReactNode } from 'react';
import React, { Children, cloneElement, isValidElement } from 'react';
import { render } from '@testing-library/react';

// NOTE: This is a temporary hack way to test server components while waiting for updates in the future.
// Reference: https://gist.github.com/sroebert/a04ca6e0232a4a60bc50d7f164f101f6
function setFakeReactDispatcher<T>(action: () => T): T {
  /**
   * We use some internals from React to avoid a lot of warnings in our tests when faking
   * to render server components. If the structure of React changes, this function should still work,
   * but the tests will again print warnings.
   *
   * If this is the case, this function can also simply be removed and all tests should still function.
   */

  if (!('__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED' in React)) {
    return action();
  }

  const secret = React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
  if (
    !secret ||
    typeof secret !== 'object' ||
    !('ReactCurrentDispatcher' in secret)
  ) {
    return action();
  }

  const currentDispatcher = secret.ReactCurrentDispatcher;
  if (
    !currentDispatcher ||
    typeof currentDispatcher !== 'object' ||
    !('current' in currentDispatcher)
  ) {
    return action();
  }

  const previousDispatcher = currentDispatcher.current;
  try {
    currentDispatcher.current = new Proxy(
      {},
      {
        get() {
          throw new Error('This is a client component');
        },
      },
    );
  } catch {
    return action();
  }

  const result = action();

  currentDispatcher.current = previousDispatcher;

  return result;
}

async function evaluateServerComponent(
  node: ReactElement,
): Promise<ReactElement> {
  if (node && node.type?.constructor?.name === 'AsyncFunction') {
    // Handle async server nodes by calling await.

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-expect-error
    const evaluatedNode: ReactElement = await node.type({ ...node.props });
    return evaluateServerComponent(evaluatedNode);
  }

  if (node && node.type?.constructor?.name === 'Function') {
    try {
      return setFakeReactDispatcher(() => {
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-expect-error
        const evaluatedNode: ReactElement = node.type({ ...node.props });
        return evaluateServerComponent(evaluatedNode);
      });
    } catch {
      // If evaluating fails with a function node, it might be because of using client side hooks.
      // In that case, simply return the node, it will be handled by the react testing library render function.
      return node;
    }
  }

  return node;
}

async function evaluateServerComponentAndChildren(node: ReactElement) {
  const evaluatedNode = await evaluateServerComponent(node);

  if (!evaluatedNode?.props.children) {
    return evaluatedNode;
  }

  const children = Children.toArray(evaluatedNode.props.children);
  for (let i = 0; i < children.length; i += 1) {
    const child = children[i];
    if (!isValidElement(child)) {
      continue;
    }

    children[i] = await evaluateServerComponentAndChildren(child);
  }

  return cloneElement(evaluatedNode, {}, ...children);
}

// Follow <https://github.com/testing-library/react-testing-library/issues/1209>
// for the latest updates on React Testing Library support for React Server
// Components (RSC)
export async function renderServerComponent(
  nodeOrPromise: ReactNode | Promise<ReactNode>,
) {
  const node = await nodeOrPromise;

  if (isValidElement(node)) {
    const evaluatedNode = await evaluateServerComponentAndChildren(node);
    return render(evaluatedNode);
  }

  return render(node);
}
