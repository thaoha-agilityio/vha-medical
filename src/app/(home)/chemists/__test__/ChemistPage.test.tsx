import { renderServerComponent } from '@/utils/test-util';
import ChemistPage from '../page';

jest.mock('next-auth', () => ({
  __esModule: true,
  default: jest.fn().mockReturnValue({
    handlers: jest.fn(),
    signOut: jest.fn(),
    auth: jest.fn(),
  }),
}));

describe('Chemists Page test cases', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', async () => {
    const { asFragment } = await renderServerComponent(<ChemistPage />);

    expect(asFragment()).toMatchSnapshot();
  });
});
