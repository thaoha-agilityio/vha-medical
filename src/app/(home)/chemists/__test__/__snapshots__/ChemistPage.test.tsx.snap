// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chemists Page test cases should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col mt-8 md:flex-row gap-4 md:mb-3"
  >
    <div
      class="group flex flex-col data-[hidden=true]:hidden w-full relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
      data-filled="true"
      data-filled-within="true"
      data-slot="base"
    >
      <div
        class="flex flex-col h-fit lg:max-w-[400px] text-xs"
        data-slot="main-wrapper"
      >
        <div
          class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background bg-transparent border-text-foreground !bg-transparent data-[hover=true]:!bg-transparent group-data-[focus=true]:ring-0 group-data-[focus=true]:ring-offset-0 group-data-[focus=true]:!bg-transparent data-[focus=true]:border-secondary-300 group-data-[invalid=true]:border-danger-100 py-4 border h-[52px]"
          data-slot="input-wrapper"
          style="cursor: text;"
        >
          <div
            class="inline-flex w-full items-center h-full box-border"
            data-slot="inner-wrapper"
          >
            <svg
              class="w-5 h-5 ml-4 text-primary-200"
              fill="none"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_1_18701)"
              >
                <path
                  d="M9.58341 17.5C13.9557 17.5 17.5001 13.9556 17.5001 9.58335C17.5001 5.2111 13.9557 1.66669 9.58341 1.66669C5.21116 1.66669 1.66675 5.2111 1.66675 9.58335C1.66675 13.9556 5.21116 17.5 9.58341 17.5Z"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
                <path
                  d="M18.3334 18.3334L16.6667 16.6667"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_1_18701"
                >
                  <rect
                    fill="white"
                    height="20"
                    width="20"
                  />
                </clippath>
              </defs>
            </svg>
            <input
              aria-label="Search Chemists"
              class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text peer pr-6 rtl:pr-0 rtl:pl-6 group-data-[has-value=true]:text-default-foreground text-primary-100 placeholder:text-primary-300 bg-transparent text-sm mx-2 group-data-[invalid=true]:!text-danger-100"
              data-has-start-content="true"
              data-slot="input"
              id="react-aria-:r0:"
              placeholder="Search Chemists"
              type="text"
              value=""
            />
            <span
              aria-label="clear input"
              class="p-2 -m-2 z-10 hidden absolute end-3 start-auto appearance-none select-none opacity-0 hover:!opacity-100 cursor-pointer active:!opacity-70 rounded-full outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-large peer-data-[filled=true]:opacity-70 peer-data-[filled=true]:block transition-opacity motion-reduce:transition-none"
              data-slot="clear-button"
              role="button"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="1em"
                role="presentation"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex justify-between md:gap-4 mb-10 md:mb-0 "
    >
      <div
        class="group flex flex-col w-full transition-background motion-reduce:transition-none !duration-150 group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_10px)]"
        data-filled="true"
        data-slot="base"
      >
        <div
          aria-hidden="true"
          data-a11y-ignore="aria-hidden-focus"
          data-testid="hidden-select-container"
          style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
        >
          <input
            style="font-size: 16px;"
            tabindex="0"
            type="text"
            value=""
          />
          <label>
            <select
              size="1"
              tabindex="-1"
              title=""
            >
              <option />
              <option
                value="all"
              >
                All
              </option>
            </select>
          </label>
        </div>
        <div
          class="w-full flex flex-col"
          data-slot="mainWrapper"
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Select Specialty"
            aria-labelledby="react-aria-:rd: react-aria-:r8:"
            class="relative gap-3 inline-flex flex-row items-center shadow-sm tap-highlight-transparent group-data-[focus=true]:bg-default-100 min-h-10 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-2.5 py-2 rounded-medium bg-background-100 shadow-stack border-1 outline-offset-0 data-[focus=true]:border-green data-[focus=true]:border-1 data-[focus=true]:bg-background-100 data-[hover=true]:bg-background-100 group-data-[invalid=true]:!border-danger-100 w-[180px] h-[52px]"
            data-slot="trigger"
            id="react-aria-:r8:"
            type="button"
          >
            <div
              class="inline-flex h-full min-h-4 items-center gap-1.5 box-border text-xs leading-[18px] group-data-[invalid=true]:!text-danger-100 w-[180px]"
              data-slot="innerWrapper"
            >
              <span
                class="w-full text-left rtl:text-right truncate group-data-[has-value=true]:text-default-foreground text-xs group-data-[invalid=true]:!text-danger-100 text-primary-100 font-medium"
                data-slot="value"
                id="react-aria-:rd:"
              >
                Specialty
              </span>
            </div>
            <svg
              aria-hidden="true"
              class="absolute end-3 w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180 text-primary-100"
              data-slot="selectorIcon"
              fill="none"
              focusable="false"
              height="1em"
              role="presentation"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <p
    class="text-primary-100 font-medium text-2xl my-5"
  >
    All Chemists
  </p>
  <div
    class="flex flex-col items-center min-h-[40vh] sm:min-h-[60vh]"
  >
    <p
      class="text-primary-200 font-medium text-lg my-auto"
    >
      Result Not Found
    </p>
  </div>
</DocumentFragment>
`;
